stores/libsql/src/storage/index.ts

```diff
@@ -32,20 +32,49 @@ function safelyParseJSON(jsonString: string): any {
 export interface LibSQLConfig {
   url: string;
   authToken?: string;
+  /**
+   * Maximum number of retries for write operations if an SQLITE_BUSY error occurs.
+   * @default 5
+   */
+  maxRetries?: number;
+  /**
+   * Initial backoff time in milliseconds for retrying write operations on SQLITE_BUSY.
+   * The backoff time will double with each retry (exponential backoff).
+   * @default 100
+   */
+  initialBackoffMs?: number;
 }
 
 export class LibSQLStore extends MastraStorage {
   private client: Client;
+  private readonly maxRetries: number;
+  private readonly initialBackoffMs: number;
 
   constructor(config: LibSQLConfig) {
     super({ name: `LibSQLStore` });
 
+    this.maxRetries = config.maxRetries ?? 5;
+    this.initialBackoffMs = config.initialBackoffMs ?? 100;
+
     // need to re-init every time for in memory dbs or the tables might not exist
     if (config.url.endsWith(':memory:')) {
       this.shouldCacheInit = false;
     }
 
     this.client = createClient(config);
+
+    // Set PRAGMAs for better concurrency, especially for file-based databases
```

Comment:
Can you move this to a private function an call it to keep constructor easy to read.